docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,387 @@
+
+## Upload a new training script or new version
+
+You must use the [Viam CLI](/cli/) to upload your training script to the Registry.
+Follow the instructions in the [CLI documentation](/cli/) to install the CLI.
+
+You can use the [`viam training-script upload`](/cli/#training-script) command to upload a new script.
+
+For example:
+
+```sh {class="line-numbers linkable-line-numbers"}
+viam training-script upload --path=<path-to-tar.gz> --org-id=<INSERT ORG ID> --script-name="MyCustomTrainingScript"
+```
+
+To find your organization's ID, navigate to your organization's **Settings** page in [the Viam app](https://app.viam.com/).
+Find **Organization ID** and click the copy icon.
+
+Once uploaded, you can view the script by navigating to the [registry's **Training Scripts** page](https://app.viam.com/registry?type=Training+Script).
+
+You can also simultaneously upload a training script and submit a training job with the [`viam train submit custom from-upload` command](/cli/#position-arguments-submit-custom).
+
+## Submit a training job
+
+You can use the Viam CLI's [`viam train submit`](/cli/#positional-arguments-submit) command to submit a training job.
+
+Referencing the [CLI documentation](/cli/#positional-arguments-submit), use `viam train submit custom from-registry` to submit a training job from a training script already uploaded to the registry and `viam train submit custom from-upload` to upload a training script and submit a training job at the same time.
+
+For example:
+
+{{< tabs >}}
+{{% tab name="from-registry" %}}
+
+```sh {class="line-numbers linkable-line-numbers"}
+viam train submit custom from-registry --dataset-id=<INSERT DATASET ID> --org-id=<INSERT ORG ID> --model-name="MyRegistryModel" --model-version="2" --version="1" --path=<path-to-tar.gz> --script-name="MyCustomTrainingScript"
+```
+
+This command submits a training job to the previously uploaded `"MyCustomTrainingScript"` with another input dataset, which trains `"MyRegistryModel"` and publishes that to the registry.
+
+{{% /tab %}}
+{{% tab name="with-upload" %}}
+
+```sh {class="line-numbers linkable-line-numbers"}
+viam train submit custom with-upload --dataset-id=<INSERT DATASET ID> --model-org-id=<INSERT ORG ID> --model-name="MyRegistryModel" --model-type="single_label_classification" --model-version="2" --version="1" --path=<path-to-tar.gz> --script-name="MyCustomTrainingScript"
+```
+
+This command uploads a script called `"MyCustomTrainingScript"` to the registry under the specified organization and also submits a training job to that script with the input dataset, which generates a new version of the single-classification ML model `"MyRegistryModel"` and publishes that to the registry.
+
+{{% /tab %}}
+{{< /tabs >}}
+
+To find the dataset ID of a given dataset, go to the [**DATASETS** subtab](https://app.viam.com/data/datasets) of the **DATA** tab on the Viam app and select a dataset.
+Click **...** in the left-hand menu and click **Copy dataset ID**.
```

Comment:
Still missing the next steps section?

Reply:
Sorry I've meant to do this for a bit but keep getting caught up in other feedback

Reply:
did not forget about it